applications/ParticleMechanicsApplication/custom_utilities/particle_mechanics_math_utilities.h

```diff
@@ -71,6 +71,84 @@ class ParticleMechanicsMathUtilities
+
+            double aux = rNormalVector[0]*rNormalVector[0] + rNormalVector[1]*rNormalVector[1];
+            aux = sqrt(aux);
+            if (std::abs(aux) < std::numeric_limits<double>::epsilon()) aux = std::numeric_limits<double>::epsilon();
+
+            rRotationMatrix(0,0) =  rNormalVector[0]/aux;
+            rRotationMatrix(0,1) =  rNormalVector[1]/aux;
+            rRotationMatrix(1,0) = -rNormalVector[1]/aux;
+            rRotationMatrix(1,1) =  rNormalVector[0]/aux;
+        }
+        else if (Dimension == 3){
+            if (rRotationMatrix.size1() != 3 && rRotationMatrix.size2() != 3)
+                rRotationMatrix.resize(2,2,false);
+            noalias(rRotationMatrix) = IdentityMatrix(Dimension);
+
+            double aux = rNormalVector[0]*rNormalVector[0] + rNormalVector[1]*rNormalVector[1] + rNormalVector[2]*rNormalVector[2];
+            aux = sqrt(aux);
+            if (std::abs(aux) < std::numeric_limits<double>::epsilon()) aux = std::numeric_limits<double>::epsilon();
+
+            rRotationMatrix(0,0) = rNormalVector[0]/aux;
+            rRotationMatrix(0,1) = rNormalVector[1]/aux;
+            rRotationMatrix(0,2) = rNormalVector[2]/aux;
+
+            // Define the new coordinate system, where the first vector is aligned with the normal
+            // To choose the remaining two vectors, we project the first component of the cartesian base to the tangent plane
+            Vector rT1 = ZeroVector(3);
+            rT1(0) = 1.0;
+            rT1(1) = 0.0;
+            rT1(2) = 0.0;
+            double dot = rRotationMatrix(0,0); //this->Dot(rN,rT1);
+
+            // It is possible that the normal is aligned with (1,0,0), resulting in norm(rT1) = 0
+            // If this is the case, repeat the procedure using (0,1,0)
+            if ( fabs(dot) > 0.99 )
+            {
+                rT1(0) = 0.0;
+                rT1(1) = 1.0;
+                rT1(2) = 0.0;
+
+                dot = rRotationMatrix(0,1); //this->Dot(rN,rT1);
+            }
+
+            // Calculate projection and normalize
+            rT1[0] -= dot*rRotationMatrix(0,0);
+            rT1[1] -= dot*rRotationMatrix(0,1);
+            rT1[2] -= dot*rRotationMatrix(0,2);
+            ParticleMechanicsMathUtilities<double>::Normalize(rT1);
+
+            rRotationMatrix(1,0) = rT1[0];
+            rRotationMatrix(1,0) = rT1[1];
+            rRotationMatrix(1,0) = rT1[2];
+
+            // The third base component is choosen as N x T1, which is normalized by construction
+            rRotationMatrix(2,0) = rRotationMatrix(0,1)*rT1[2] - rRotationMatrix(0,2)*rT1[1];
+            rRotationMatrix(2,1) = rRotationMatrix(0,2)*rT1[0] - rRotationMatrix(0,0)*rT1[2];
+            rRotationMatrix(2,2) = rRotationMatrix(0,0)*rT1[1] - rRotationMatrix(0,1)*rT1[0];
+        }
+        else{
+            KRATOS_ERROR <<  "Dimension given is wrong: Something is wrong with the given dimension in function: GetRotationMatrix" << std::endl;
```

Comment:
```suggestion
            KRATOS_ERROR <<  "Dimension given is wrong" << std::endl;
```

the location is automatically printed